### PR TITLE
Replace node-latex with Tectonic

### PR DESCRIPTION
**Pull request by ducaale**
Resolves https://github.com/saadq/resumake.io/issues/179

### CODE REVIEW

File: src/utils/latex.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  tectonic.stdout.on('data', data => {
+    console.log('tectonic stdout:', data.toString().trim())
+  })
+  tectonic.stderr.on('data', data => {
+    console.error('tectonic stderr:', data.toString().trim())
+  })
+  await new Promise<void>((resolve, reject) => {
+    tectonic.on('close', (code) => {
+      if (code !== 0) {
+        console.log(`tectonic exited with code ${code}`);
```
Comment (ducaale) [inline src/utils/latex.ts:22]: It would be nice to setup CI that catches any code style issues

**Comment by devdrowsy** [comment]
Thanks! Will be able to review this once I'm off my flight on Saturday

File: src/utils/latex.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { spawn } from 'node:child_process'
+import { mkdtemp } from 'node:fs/promises'
+import { createReadStream } from 'node:fs'
+import os from 'node:os'
+import path from 'node:path'
```
Comment (devdrowsy) [inline src/utils/latex.ts:5]: Haven't seen this `node:*` style import before, do you have a reference for it?

**Review by devdrowsy — APPROVED** [review]
LGTM, thanks for this change!

File: src/utils/latex.ts (hunk 2 above)
Comment (ducaale) [reply src/utils/latex.ts:5]: This is something I learned from https://twitter.com/rauschma/status/1516456272215826432

The [docs](https://nodejs.org/api/modules.html) say that support for the core module prefix was added in `v16.0.0` and `v14.18.0`

**Pull request merged**